partiql-parser/src/lalr/partiql.lalrpop

```diff
@@ -0,0 +1,584 @@
+//                            UNION/INTERSECT/EXCEPT                              //
+// ------------------------------------------------------------------------------ //
+// TODO
+SetOpClause: () = {}
+
+// ------------------------------------------------------------------------------ //
+//                                   ORDER BY                                     //
+// ------------------------------------------------------------------------------ //
+OrderByClause: Box<ast::OrderByExpr> = {
+    "ORDER" "BY" "PRESERVE" => Box::new( ast::OrderByExpr{ sort_specs: vec![] } ),
+    "ORDER" "BY" <sort_specs: CommaSep<OrderSortSpec>> => Box::new( ast::OrderByExpr{ sort_specs } ),
+}
+OrderSortSpec: ast::SortSpec = {
+    <expr:ExprQuery> <ordering_spec:BySpec?> <null_ordering_spec:ByNullSpec?>
+        => ast::SortSpec { expr, ordering_spec, null_ordering_spec }
+}
+BySpec: ast::OrderingSpec = {
+    "ASC" => ast::OrderingSpec{ kind: ast::OrderingSpecKind::Asc },
+    "DESC" => ast::OrderingSpec{ kind: ast::OrderingSpecKind::Desc },
+}
+ByNullSpec: ast::NullOrderingSpec = {
+    "NULLS" "FIRST" => ast::NullOrderingSpec{ kind: ast::NullOrderingSpecKind::First },
+    "NULLS" "LAST" => ast::NullOrderingSpec{ kind: ast::NullOrderingSpecKind::First },
+}
+
+// ------------------------------------------------------------------------------ //
+//                                     LIMIT                                      //
+// ------------------------------------------------------------------------------ //
+LimitClause: Box<ast::Expr> = { "LIMIT" <ExprQuery> }
+
+// ------------------------------------------------------------------------------ //
+//                                     OFFSET                                     //
+// ------------------------------------------------------------------------------ //
+OffsetByClause: Box<ast::Expr> = { "OFFSET" <ExprQuery> }
+
+// ------------------------------------------------------------------------------ //
+//                                                                                //
+//                                   Expr Query                                   //
+//                                                                                //
+// ------------------------------------------------------------------------------ //
+
+pub ExprQuery : Box<ast::Expr> = {
+    <e:ExprPrec15> => Box::new(e)
+}
+
+ExprPrec15 : ast::Expr = {
+    <l:ExprPrec15> "OR" <r:ExprPrec14> =>
+       ast::Expr{ kind: ast::ExprKind::Or( ast::Or{ operands: vec![Box::new(l),Box::new(r)] } ) },
+    <ExprPrec14>,
+}
+
+ExprPrec14 : ast::Expr = {
+    <l:ExprPrec14> "AND" <r:ExprPrec13> =>
+       ast::Expr{ kind: ast::ExprKind::And( ast::And{ operands: vec![Box::new(l),Box::new(r)] } ) },
+    <ExprPrec13>,
+}
+
+ExprPrec13 : ast::Expr = {<ExprPrec12>,}
+ExprPrec12 : ast::Expr = {<ExprPrec11>,}
+ExprPrec11 : ast::Expr = {<ExprPrec10>,}
+ExprPrec10 : ast::Expr = {
```

Comment:
Irrespective of which precedence model we choose, does it make sense to leave some room between ranks? (E.g. 10, 20, ... )
I understand once we set them, we're good, but since we're at the early stages might make sense to leave room for some other operators that we might add in the future.